.github/workflows/build_and_tests.yml

```diff
@@ -0,0 +1,37 @@
+name: Build H2M-Launcher and run tests
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+
+jobs:
+  build-and-tests:
+    name: Build and run tests
+    runs-on: windows-latest
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Setup .NET
+        uses: actions/setup-dotnet@v4
+        with:
+          dotnet-version: 8.x
+
+      - name: Build project
+        run: dotnet build H2M-Launcher.sln --configuration Debug
```

Comment:
Is there a specific reason to build with debug configuration? I would rather prefer Release because there will be some differences that affect the build.

Reply:
I can't explain why, but it I use ``--configuration Release``, the test cannot be executed.
See this PR where I tried to use ``--configuration Release``: https://github.com/moi15moi/H2M-Launcher/pull/2/checks

I'm not really familiar with dotnet, but if you have an idea to resolve this problem

Reply:
I am pretty sure you need to pass the configuration to the test command too. It seems to look for the Debug assembly by default

Reply:
Ok, done.
But, if you want, I can replace ``dotnet test H2M-Launcher.sln --configuration Release --no-build --verbosity normal`` with ``dotnet test H2M-Launcher.sln --configuration Release --verbosity normal`` which will build the project and run the test in 1 step instead of 2

Reply:
No i think the division is good 👍